### DISCUSSION TITLE
My hotword's perfomance is bad

### DISCUSSION THREAD

**Discussion by TheSeriousProgrammer** [Ant-Brain/EfficientWord-Net#4] [Q&A]
If you are playing around with using your own custom hotwords and some hotword happen to not work so good
Feel free to use this thread

Make sure that you give us the following information:

- Basic Inference Code you use
- attach a .zip containing your xyz_ref.json file , sample audio files used to generate the reference 

Note that certain hotwords might not give good perfomance when you try to generalize it
In such scenarios you can try to collect samples directly from the target user
or use a different hotword (lol)

**Comment by ghost** [comment]
I am having a problem creating the xyz_ref.json file it shows the following error:


Paste Path of folder Containing audio files:P:\audio file\veronika
Paste Path of location to save *_ref.json :P:\Project Veronica
Enter Wakeword Name                                 :veronika
C:\Users\Dell\AppData\Local\Programs\Python\Python38\lib\site-packages\librosa\util\decorators.py:88: UserWarning: PySoundFile failed. Trying audioread instead.
  return f(*args, **kwargs)
Traceback (most recent call last):
  File "C:\Users\Dell\AppData\Local\Programs\Python\Python38\lib\site-packages\librosa\core\audio.py", line 164, in load    y, sr_native = __soundfile_load(path, offset, duration, dtype)
  File "C:\Users\Dell\AppData\Local\Programs\Python\Python38\lib\site-packages\librosa\core\audio.py", line 195, in __soundfile_load
    context = sf.SoundFile(path)
  File "C:\Users\Dell\AppData\Local\Programs\Python\Python38\lib\site-packages\soundfile.py", line 629, in __init__
    self._file = self._open(file, mode_int, closefd)
  File "C:\Users\Dell\AppData\Local\Programs\Python\Python38\lib\site-packages\soundfile.py", line 1183, in _open
    _error_check(_snd.sf_error(file_ptr),
  File "C:\Users\Dell\AppData\Local\Programs\Python\Python38\lib\site-packages\soundfile.py", line 1357, in _error_check    raise RuntimeError(prefix + _ffi.string(err_str).decode('utf-8', 'replace'))
RuntimeError: Error opening 'P:\\audio file\\veronika\\veronika_en-GB_CharlotteV3Voice.mp3': File contains data in an unknown format.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Dell\AppData\Local\Programs\Python\Python38\lib\runpy.py", line 192, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\Dell\AppData\Local\Programs\Python\Python38\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\Dell\AppData\Local\Programs\Python\Python38\lib\site-packages\eff_word_net\generate_reference.py", line 77, in <module>
    generate_reference_file(
  File "C:\Users\Dell\AppData\Local\Programs\Python\Python38\lib\site-packages\eff_word_net\generate_reference.py", line 47, in generate_reference_file
    x,_ = librosa.load(audio_file,sr=16000)
  File "C:\Users\Dell\AppData\Local\Programs\Python\Python38\lib\site-packages\librosa\util\decorators.py", line 88, in inner_f
    return f(*args, **kwargs)
  File "C:\Users\Dell\AppData\Local\Programs\Python\Python38\lib\site-packages\librosa\core\audio.py", line 170, in load    y, sr_native = __audioread_load(path, offset, duration, dtype)
  File "C:\Users\Dell\AppData\Local\Programs\Python\Python38\lib\site-packages\librosa\core\audio.py", line 226, in __audioread_load
    reader = audioread.audio_open(path)
  File "C:\Users\Dell\AppData\Local\Programs\Python\Python38\lib\site-packages\audioread\__init__.py", line 132, in audio_open
    raise NoBackendError()
audioread.exceptions.NoBackendError

**Reply by TheSeriousProgrammer** [reply]
consider converting your mp3 files to wav files using ffmpeg, and feeding it as input. This will help in resolving the same

> I am having a problem creating the xyz_ref.json file it shows the following error:
> 
> Paste Path of folder Containing audio files:P:\audio file\veronika Paste Path of location to save *_ref.json :P:\Project Veronica Enter Wakeword Name :veronika C:\Users\Dell\AppData\Local\Programs\Python\Python38\lib\site-packages\librosa\util\decorators.py:88: UserWarning: PySoundFile failed. Trying audioread instead. return f(*args, **kwargs) Traceback (most recent call last): File "C:\Users\Dell\AppData\Local\Programs\Python\Python38\lib\site-packages\librosa\core\audio.py", line 164, in load y, sr_native = __soundfile_load(path, offset, duration, dtype) File "C:\Users\Dell\AppData\Local\Programs\Python\Python38\lib\site-packages\librosa\core\audio.py", line 195, in __soundfile_load context = sf.SoundFile(path) File "C:\Users\Dell\AppData\Local\Programs\Python\Python38\lib\site-packages\soundfile.py", line 629, in **init** self._file = self._open(file, mode_int, closefd) File "C:\Users\Dell\AppData\Local\Programs\Python\Python38\lib\site-packages\soundfile.py", line 1183, in _open _error_check(_snd.sf_error(file_ptr), File "C:\Users\Dell\AppData\Local\Programs\Python\Python38\lib\site-packages\soundfile.py", line 1357, in _error_check raise RuntimeError(prefix + _ffi.string(err_str).decode('utf-8', 'replace')) RuntimeError: Error opening 'P:\audio file\veronika\veronika_en-GB_CharlotteV3Voice.mp3': File contains data in an unknown format.
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last): File "C:\Users\Dell\AppData\Local\Programs\Python\Python38\lib\runpy.py", line 192, in _run_module_as_main return _run_code(code, main_globals, None, File "C:\Users\Dell\AppData\Local\Programs\Python\Python38\lib\runpy.py", line 85, in _run_code exec(code, run_globals) File "C:\Users\Dell\AppData\Local\Programs\Python\Python38\lib\site-packages\eff_word_net\generate_reference.py", line 77, in generate_reference_file( File "C:\Users\Dell\AppData\Local\Programs\Python\Python38\lib\site-packages\eff_word_net\generate_reference.py", line 47, in generate_reference_file x,_ = librosa.load(audio_file,sr=16000) File "C:\Users\Dell\AppData\Local\Programs\Python\Python38\lib\site-packages\librosa\util\decorators.py", line 88, in inner_f return f(*args, **kwargs) File "C:\Users\Dell\AppData\Local\Programs\Python\Python38\lib\site-packages\librosa\core\audio.py", line 170, in load y, sr_native = __audioread_load(path, offset, duration, dtype) File "C:\Users\Dell\AppData\Local\Programs\Python\Python38\lib\site-packages\librosa\core\audio.py", line 226, in __audioread_load reader = audioread.audio_open(path) File "C:\Users\Dell\AppData\Local\Programs\Python\Python38\lib\site-packages\audioread__init__.py", line 132, in audio_open raise NoBackendError() audioread.exceptions.NoBackendError

**Reply by ghost** [reply]
Thank you so much, this method works.

**Comment by konek3** [comment]
Hello!

I have a problem with EfficientWord-Net, when I'm initializing it it works just fine, but it has a lot of false positives when I'm clicking my mouse near microphone - no matter if I use sample hotword like Alexa or MyCroft or my custom one. What might be the problem? Confidence level on clicking mouse is sometimes even 0.97 which is very high and makes hotword detection unusable...

**Reply by TheSeriousProgrammer** [reply]
Thanks for letting us know on the same, will definitely have a look at it

I was not able to reproduce your issue, can you help by sharing a video recording of the same

Can you try and see if the same is happening with a custom wakeword designed from your voice samples?

**Reply by yongliang-wu** [reply]
Hello, I have the same problem. And I record the video, and i update it to [google drive](https://drive.google.com/file/d/1dodZu3Sb54J-kIYxRZnNVqb7Le9cfXcm/view?usp=share_link). I use python=3.9.13, pyaudio=0.2.11, tflite=2.5.0.

**Comment by 3sticks** [comment]
Hey, this library is incredible. 

Using the ibm generator I get pretty good results, but I wanted to see if I could train it on my own voice. I set up 9 .wav recordings of me saying "office" various ways (and from various distances), but when I ran it, every noise picked up was triggered at >93%. Snapping my fingers in the mic was 99%. I figured maybe it was a .wav issue so I converted everything to mp3, but still no dice. 

Zip is attached, used your generator and the suggested hotword code when running script. 

I am only bringing this up because I am curious what the difference between my voice files and the IBM ones. seriously, this library is amazing, you guys did incredible work. 

[office_.zip](https://github.com/Ant-Brain/EfficientWord-Net/files/10471385/office_.zip)

**Reply by 3sticks** [reply]
Okay reading through some other people's issues, I see you mention the model is built for 1 second long audio clips, and mine are 2 seconds. I will rerun with 1 second long files, and let you know if I have the same issue.

**Comment by 97Cweb** [comment]
Looking through this project seems to be exactly what I need for my voice assistant I am building. The main STT I am using is designed for longer real sentences while this is for short wakewords. 
However, I tried using this library, but not even the default config works. I am not sure what is erroring out. Also, the IBM generator no longer works. I am running it on a raspberry pi.
[Wakeword.zip](https://github.com/Ant-Brain/EfficientWord-Net/files/11009678/Wakeword.zip)

If it helps, I also tried the my voice option, and I get NaN errors. It might be due to the fact I edited them in audacity after the first json did not work, but I don't know.
Can you take a look at them?

[Wakeword.zip](https://github.com/Ant-Brain/EfficientWord-Net/files/11009678/Wakeword.zip)
Other json file that shows NAN
[HeySox_ref.zip](https://github.com/Ant-Brain/EfficientWord-Net/files/11009683/HeySox_ref.zip)

**Comment by jeevajoji** [comment]
if you dont mind please share the code for resnet with arc loss .
